### ISSUE TITLE
Any plans for migration of DTF to .NET Core? 

### ISSUE THREAD

**Issue by ddobric** [Azure/durabletask#96]
Any plans for migration of DTF to .NET Core?

**Comment by simonporter** [comment]
.NET Core support is planned, the vnext branch is a starting point to this end, in the new csproj format.
DTF has a dependency on ServiceBus which does not yet have a .NET Core package will all the required features.

**Comment by SeanFeldman** [comment]
Once https://github.com/Azure/azure-service-bus-dotnet/issues/6 is resolved, this one will be unblocked.

**Comment by ElanHasson** [comment]
i think it should be unblocked-- System.Transactions has been released.

I started this work and it seems the major issue is around the breaking changes in Azure Service Bus .net core port which has a HUGE api surface change, and conceptual changes.

There is also a lack of documentation on how to address those changes.

**Comment by ElanHasson** [comment]
- [X] DurableTask.Core
- [X] DurableTask.Emulator
- [ ]  DurableTask.ServiceBus
  - [ ] MessageSession is now internal, => use IMessageSession
  - [ ]  MessagingFactory => new paradigm
  - [ ]  NamespaceManager => new paradigm
  - [X]  Methods that required LockTokens as parameters accepted System.Guid => now they take string; replaced with lockToken.ToString("N")
  - [X]  BrokeredMessage => Message
  - [] Message.EnqueuedTimeUtc => ?
  - [] Message.LockedUntilUtc => ?
  - [] ctor for Message no longer has an override that takes object, only byte[]
  - [] No longer using Streams to work with messages, uses only byte[]
  - [] Message.DeliveryCount => ?
  - [] Message.EnqueuedTimeUtc => ?
  - [] QueueDescription => ?
  - [] QueueClient.AcceptMessageSessionAsync => ?
  - [] Message.SequenceNumber=> ?
  - [] Message.LockToken => ?
  - [] IMessageSession does not have an Abort() => ?
  - [] QueueClient.ReceiveAsync() => ?
  - [] Message.RenewLockAsync() => ? 
  - [] Message.AbandonAsync() => ? 
  - [] MessagingException => ?
  - [] MessagingEntityAlreadyExistsException => ?

This covers the work for DurableTask.ServiceBus.  I stopped here because this is a core dependency in the project presently. 

Obviously a sample need to be addressed because the use WinForms.

**Comment by SeanFeldman** [comment]
@ElanHasson Management operations will **not** be provided by the new ASB client.

- `QueueDescription` doesn't map to anything<sup>1</sup>.
- `MessagingException` => `ServiceBusException`
- `MessagingEntityAlreadyExistsException` doesn't map to anything<sup>1</sup>.

Also

> I think it should be unblocked-- System.Transactions has been released

ASB client won't support transactions semantics [until 2.0.0](https://github.com/Azure/azure-service-bus-dotnet/issues/6).

Hope that helps.

<sup>1</sup> See details see https://github.com/Azure/azure-service-bus-dotnet#can-i-manage-service-bus-entities-with-this-library

**Comment by ElanHasson** [comment]
Thanks Sean.

I've already started working on the implementation of the management stuff.

Are transactions required my durable task? Or only for some features.

Also, it seems Azure Service Bus isn't getting a lot of love lately. Perhaps a new provider with EventHub/EventGrid?

**Comment by SeanFeldman** [comment]
> Are transactions required my durable task? Or only for some features.

It's only required if you want your incoming and outgoing messages to participate in a transaction.

> Also, it seems Azure Service Bus isn't getting a lot of love lately.

Not sure what do you mean by that. GA-ed 1.0 of the new client is not enough?

**Comment by simonporter** [comment]
DurableTask does rely heavily in transactions to synchronize setting the session state, orchestration messages and activities using send-via

**Comment by SeanFeldman** [comment]
Transactions semantics will be supported with the new ASB client from 2.0.0

**Comment by cgillum** [comment]
How about the non-Service Bus parts of the framework (for example, DurableTask.Core)? Is work being done today to port that over to .NET Core?

In my case, I don't use the Service Bus package. Rather, I have an [Azure Storage provider implementation](https://github.com/Azure/durabletask/tree/vnext_azurestorage/src/DurableTask.AzureStorage) which will need to be ported to .NET Core soon (to support Azure Functions 2.0). The discussion so far has been focused on Service Bus, but can I assume there aren't any fundamental blockers with porting over the core bits?

Obviously this will need to be done either way, but I'm just curious if any effort has already started.

**Comment by cgillum** [comment]
Ah, nevermind. I see that the checklist above already includes DurableTask.Core (as does the PR). I should be in good shape then. :)

**Comment by VenkateshSrini** [comment]
Do you have a release date by which Durable Task frame work for .NET core will be available ?

**Comment by simonporter** [comment]
DurableTask.Core, DurableTask.Emulator and DurableTask.AzureStorage are all now published for NetStandard2.0.
Need to revisit the ServiceBus work assuming transaction support has been added else consider a refactoring that does not require transactions.

**Comment by SeanFeldman** [comment]
No transactions support for ASB yet.

**Comment by malafreniere** [comment]
Any news on this ?

Thank you

**Comment by mnadeau2** [comment]
any news ?

**Comment by MisinformedDNA** [comment]
Is the plan to target .NET Core or .NET Standard? I ask because new Azure Function projects target .NET Standard. See https://github.com/Azure/Azure-Functions/issues/686#issuecomment-389593840

**Comment by Preecington** [comment]
Following up on my last comment - I’ve asked a bunch of questions for guidance on my posted PR but it seems to have been forgotten? I’d really like to close it out and get it merged, so I’d love to get some feedback on the questions posted. 👍

**Comment by malafreniere** [comment]
Any news ? Can't wait for the PR to be merged :)

**Comment by keodime** [comment]
Same here ! :D

**Comment by Preecington** [comment]
I'm working very hard on building a solution that satisfies the comments @simonporter and others have made - I really hope to have something to show soon!

**Comment by Preecington** [comment]
Just dropping a reminder here to @simonporter and others that the PR is awaiting review for the dual Fx/Standard implementation.

**Comment by lapellaniz** [comment]
What's the ETA on this change? .NET Standard support is required for my use case.

**Comment by MisinformedDNA** [comment]
Isn't this already done? [Microsoft.Azure.DurableTask.Core](https://www.nuget.org/packages/Microsoft.Azure.DurableTask.Core) supports .NET Standard 2.0.

**Comment by Preecington** [comment]
That's true @MisinformedDNA but [Microsoft.Azure.DurableTask.ServiceBus](https://www.nuget.org/packages/Microsoft.Azure.DurableTask.ServiceBus) does not yet. Merged PR #203 should fix up the discrepancy soon!

**Comment by yaublet** [comment]
Looks like the branch has been merged in master. Is there a release planned soon?

**Comment by simonporter** [comment]
https://www.nuget.org/packages/Microsoft.Azure.DurableTask.ServiceBus/2.1.3 has been released with . NET standard support.

**Comment by yaublet** [comment]
Thanks Simon. 
I was looking at the release section on GitHub but looks like it's not up-to-date and maintained

**Comment by MisinformedDNA** [comment]
Close issue?

**Comment by cgillum** [comment]
Closing out old issues. .NET Core is now supported across the board.
